metaflow/plugins/airflow/airflow_cli.py

```
@@ -0,0 +1,198 @@
+import re
+
+from metaflow import S3, current, decorators
+from metaflow._vendor import click
+from metaflow.exception import MetaflowException
+from metaflow.package import MetaflowPackage
+from metaflow.plugins import EnvironmentDecorator, KubernetesDecorator
+from metaflow.util import get_username
+
+from .airflow import Airflow
+from .exception import AirflowException, NotSupportedException
+
+VALID_NAME = re.compile("[^a-zA-Z0-9_\-\.]")
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="Commands related to Airflow.")
+@click.option(
+    "--name",
+    default=None,
+    type=str,
+    help="Airflow DAG name. The flow name is used instead if this option is not "
+    "specified",
+)
+@click.pass_obj
+def airflow(obj, name=None):
+    obj.check(obj.graph, obj.flow, obj.environment, pylint=obj.pylint)
+    obj.dag_name = resolve_dag_name(obj, name)
+
+
+@airflow.command(help="Compile a new version of this workflow to Airflow DAG.")
+@click.argument("file", required=True)
+@click.option(
+    "--tag",
+    "tags",
+    multiple=True,
+    default=None,
+    help="Annotate all objects produced by Airflow DAG executions "
+    "with the given tag. You can specify this option multiple "
+    "times to attach multiple tags.",
+)
+@click.option(
+    "--is-paused-upon-creation",
+    default=False,
+    is_flag=True,
+    help="Generated Airflow DAG is paused/unpaused upon creation.",
+)
+@click.option(
+    "--namespace",
+    "user_namespace",
+    default=None,
+    # TODO (savin): Identify the default namespace?
+    help="Change the namespace from the default to the given tag. "
+    "See run --help for more information.",
+)
+@click.option(
+    "--max-workers",
+    default=100,
+    show_default=True,
+    help="Maximum number of parallel processes.",
+)
+@click.option(
+    "--workflow-timeout",
+    default=None,
+    type=int,
+    help="Workflow timeout in seconds. Enforced only for scheduled DAGs.",
```

Can we throw a pretty error to the user if we detect that they want to set the workflow timeout for an unscheduled DAG?

---

Change included in `_validate_workflow`